jujugui/static/gui/src/app/components/profile/credential-list/credential-list.js

```diff
@@ -0,0 +1,216 @@
+/* Copyright (C) 2018 Canonical Ltd. */
+'use strict';
+
+const PropTypes = require('prop-types');
+const React = require('react');
+const shapeup = require('shapeup');
+
+const BasicTable = require('../../basic-table/basic-table');
+const GenericButton = require('../../generic-button/generic-button');
+const Spinner = require('../../spinner/spinner');
+
+class ProfileCredentialList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      credentialMap: new Map(),
+      loading: true
```

Comment:
The reason why I put it here and not when we start to fetch the data is because I wanted this to be rendered with the spinner going and not flash some other UI shortly before updating to the spinner